### PR TITLE
stringnumber: radius protected, circle configurable. 

### LINKED ISSUE #1364: OSMD: String numbers colliding
> * [ ]  String numbers colliding vertically in test_grace_note_fingerings_and_strings
>   image

### PR DESCRIPTION
**Pull request by rvilarl**
fixes #1364

I added one test without circle (this one collides as described in #1364 without the changes in this PR).
StringNumber String_Number_In_Notation___radius_0 Bravura

which is a variant of the standard with radius 8
StringNumber String_Number_In_Notation Bravura

### CODE REVIEW

File: src/factory.ts [hunk 1]
```diff
@@ -464,10 +464,11 @@ export class Factory {
     return fingering;
   }
 
-  StringNumber(params: { number: string; position: string }): StringNumber {
+  StringNumber(params: { number: string; position: string; radius?: number }): StringNumber {
     const stringNumber = new StringNumber(params.number);
     stringNumber.setPosition(params.position);
     stringNumber.setContext(this.context);
+    if (params.radius != undefined) stringNumber.radius = params.radius;
```
Comment (0xfe) [inline src/factory.ts:471]: Not a fan of manipulating the instance outside an API call.

**Review by 0xfe — CHANGES_REQUESTED** [review]
I don't understand this change -- it doesn't match the PR title. Why not just disallow (or special case) radius `0`?

File: src/factory.ts (hunk 1 above)
Comment (rvilarl) [reply src/factory.ts:471]: Me neither but `radius` has no setter / getter and it is a public member. OSMD sets radius to 0 in order to present stringNumbers without circles.

What do you think about my considerations above?:

> I wonder if it really makes sense to allow to set different radius:
> 
> - Radii bigger than 8 overlap on chords and smaller than 7 overlap with the number.
> - the this.setWidth(20); // ??? is also somehow related to the radius.
> 
> I would recommend changing the class to allow the configuration of wheter a circle should be drawn.

With this I mean amking radius protected and adding a function or an option to say if the circle should be drawn.

**Comment by rvilarl** [comment]
> I don't understand this change -- it doesn't match the PR title. Why not just disallow (or special case) radius `0`?

The PR title refers to the associated issue #1346 there you can see that the string numbers (Roman numbers) collide with the notes. I think we agree, I will make radius protected and allow the special case without circle. I did not do that in the first instance because it changes the interface.

**Comment by rvilarl** [comment]
> > I don't understand this change -- it doesn't match the PR title. Why not just disallow (or special case) radius `0`?
> 
> The PR title refers to the associated issue #1364 there you can see that the string numbers (Roman numbers) collide with the notes. I think we agree, I will make radius protected and allow the special case without circle. I did not do that in the first instance because it changes the interface.

@0xfe done, ready to review.

File: src/factory.ts (hunk 1 above)
Comment (rvilarl) [reply src/factory.ts]: New approach pushed.

File: src/factory.ts [hunk 2]
```diff
@@ -464,10 +464,11 @@ export class Factory {
     return fingering;
   }
 
-  StringNumber(params: { number: string; position: string }): StringNumber {
+  StringNumber(params: { number: string; position: string; drawCircle?: boolean }): StringNumber {
```
Comment (0xfe) [inline src/factory.ts:467]: It's not clear from this if the circle is drawn or not by default. Can you make it `drawCircle: boolean = true`? (You can remove the `if undefined` below too, in that case.)

Comment (rvilarl) [reply src/factory.ts]: This has to be done outside params, done.

**Pull request merged**